### ISSUE TITLE
[BUG] 'GenericIndex' object has no attribute 'isin'

### ISSUE THREAD

**Issue by galipremsagar** [rapidsai/cudf#1905]
**Describe the bug**
I'm trying to access records at particular indices in a dask dataframe. The following code works with dask but not when used along with dask-cudf.

`gpu_ddf_selected = ddf.map_partitions(lambda x: x[x.index.isin(indices_list)])`

When I try to use `isin` in the following code I get the below mentioned exception.

**Steps/Code to reproduce bug**
Run the following code and dataframe contents mentioned below:

```
gpu_ddf_selected = ddf.map_partitions(lambda x: x[x.index.isin(indices_list)])
x = gpu_ddf_selected.persist()
wait(x)
```

Exception:

```
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/utils.py in raise_on_meta_error(funcname, udf)
    155     try:
--> 156         yield
    157     except Exception as e:

/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in _emulate(func, *args, **kwargs)
   3764     with raise_on_meta_error(funcname(func), udf=kwargs.pop('udf', False)):
-> 3765         return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
   3766 

<timed exec> in <lambda>(x)

AttributeError: 'GenericIndex' object has no attribute 'isin'

During handling of the above exception, another exception occurred:

ValueError                                Traceback (most recent call last)
<timed exec> in <module>

/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in map_partitions(self, func, *args, **kwargs)
    561         >>> ddf.map_partitions(func).clear_divisions()  # doctest: +SKIP
    562         """
--> 563         return map_partitions(func, self, *args, **kwargs)
    564 
    565     @insert_meta_param_description(pad=12)

/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in map_partitions(func, *args, **kwargs)
   3803         # Use non-normalized kwargs here, as we want the real values (not
   3804         # delayed values)
-> 3805         meta = _emulate(func, *args, udf=True, **kwargs)
   3806     else:
   3807         meta = make_meta(meta, index=meta_index)

/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py in _emulate(func, *args, **kwargs)
   3763     """
   3764     with raise_on_meta_error(funcname(func), udf=kwargs.pop('udf', False)):
-> 3765         return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
   3766 
   3767 

/conda/envs/rapids/lib/python3.7/contextlib.py in __exit__(self, type, value, traceback)
    128                 value = type()
    129             try:
--> 130                 self.gen.throw(type, value, traceback)
    131             except StopIteration as exc:
    132                 # Suppress StopIteration *unless* it's the same exception that

/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/utils.py in raise_on_meta_error(funcname, udf)
    171                 "{2}")
    172         msg = msg.format(" in `{0}`".format(funcname) if funcname else "", repr(e), tb)
--> 173         raise ValueError(msg)
    174 
    175 

ValueError: Metadata inference failed in `lambda`.

You have supplied a custom function and Dask is unable to 
determine the type of output that that function returns. 

To resolve this please provide a meta= keyword.
The docstring of the Dask function you ran should have more information.

Original error is below:
------------------------
AttributeError("'GenericIndex' object has no attribute 'isin'")

Traceback:
---------
  File "/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/utils.py", line 156, in raise_on_meta_error
    yield
  File "/conda/envs/rapids/lib/python3.7/site-packages/dask/dataframe/core.py", line 3765, in _emulate
    return func(*_extract_meta(args, True), **_extract_meta(kwargs, True))
  File "<timed exec>", line 7, in <lambda>
```
**Expected behavior**
![image](https://user-images.githubusercontent.com/11664259/58750608-36a1b080-845a-11e9-95a3-8bc5430c784b.png)

**Environment details (please complete the following information):**
 - Environment location: Docker
 - Method of cuDF install: conda-nightly
 - `cudf/print_env.sh` script output:

[env.txt](https://github.com/rapidsai/cudf/files/3243930/env.txt)

 

**Additional context**
I'm trying to achieve `take` in dask-cudf. For normal dask I was able to achieve it with the above code.

**Comment by beckernick** [comment]
This can be addressed in #2245

**Comment by kkraus14** [comment]
Fixed as of 0.9.
